trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/services/IdentifierService.java

```diff
@@ -0,0 +1,72 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Ericsson
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License 2.0 which
+ * accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.tracecompass.incubator.internal.trace.server.jersey.rest.core.services;
+
+import static org.eclipse.tracecompass.incubator.internal.trace.server.jersey.rest.core.services.EndpointConstants.VERSION;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.tags.Tag;
+
+/**
+ *
+ * Service to identify important information regarding the trace server and the system it is running on.
+ *
+ * @author Vlad Arama
+ *
+ */
+@Path("/identifier")
+@Tag(name = EndpointConstants.IDF)
+public class IdentifierService {
+    /**
+     * Getter returning important information about the system, including server version, OS, CPU count, memory size, launcher and product names.
+     *
+     * @return A JSON response containing the system's details.
+     */
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(summary = "Retrieves system and server information", responses = {
+            @ApiResponse(responseCode = "200", description = "Successfully retrieved the system and server information")
+    })
+    public Response getSystemInfo() {
+        int nbCpus = Runtime.getRuntime().availableProcessors();
+        long memory = Runtime.getRuntime().totalMemory();
+        String os = System.getProperty("os.name");
+        String version = VERSION;
+        String launcherName = System.getProperty("eclipse.launcher.name");
+        String executableName = System.getProperty("eclipse.product");
+
+        if (launcherName == null) {
+            launcherName = System.getProperty("eclipse.application");
```

Comment:
This returns the product ID: org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core.id, same as `eclipse.product` system event

Reply:
The reason for implementing this check is that I wanted a way to identify the executable when running the tests. Since `eclipse.launcher.name` is null for the unit tests and `eclipse.product = org.eclipse.platform.ide`, I added this conditional check to return `eclipse.application = org.eclipse.tycho.surefire.osgibooter.headlesstest`. It was a way to make sure that the launcherName would never be null. We can keep it like that or remove it and make launcherName optional, what do you think ?

Reply:
Launcher name is now optional 👍